src/main/scala/mesosphere/marathon/core/instance/Instance.scala

```diff
@@ -267,12 +272,16 @@ object Instance {
   case class AgentInfo(
     host: String,
     agentId: Option[String],
-    attributes: Seq[mesos.Protos.Attribute])
+    region: String,
+    zone: String,
+    attributes: Seq[Attribute])
 
   object AgentInfo {
     def apply(offer: org.apache.mesos.Protos.Offer): AgentInfo = AgentInfo(
       host = offer.getHostname,
       agentId = Some(offer.getSlaveId.getValue),
+      region = OfferUtil.region(offer).getOrElse(""),
```

Comment:
Does it make sense to make `region` and `zone` fields optional?

Reply:
I thought about this; can you think of a reason?

One thing I was considering is if region and zone are unspecified, IE, in the case of the home region, we still need to match them. Having them be None would make our matching logic more complicated (there is no way to make a constraint `region is None`, only `region is ""`)

Reply:
Thinking about it again, this may have been more valid with the implicit constraints approach. Maybe it would just be better as None. If the user wants to avoid the home region, they can specify that tasks should launch in a region where region is specified.

Reply:
What have you decided to use?

Reply:
Went with None

✅